flocker/testtools/_base.py

```diff
@@ -0,0 +1,147 @@
+# Copyright ClusterHQ Inc.  See LICENSE file for details.
+
+"""
+Base classes for unit tests.
+"""
+
+from itertools import tee
+import json
+
+from eliot.prettyprint import pretty_format
+import fixtures
+import testtools
+from testtools.content import Content, text_content
+from testtools.content_type import UTF8_TEXT
+from testtools.deferredruntest import (
+    AsynchronousDeferredRunTestForBrokenTwisted)
+
+from twisted.python.filepath import FilePath
+from twisted.trial import unittest
+
+
+class TestCase(unittest.SynchronousTestCase):
+    """
+    Base class for synchronous test cases.
+    """
+
+
+def async_runner(timeout):
+    """
+    Make a ``RunTest`` instance for asynchronous tests.
+
+    :param float timeout: The maximum length of time (in seconds) that a test
+        is allowed to take.
+    """
+    # XXX: Looks like the acceptance tests (which were the first tests that we
+    # tried to migrate) aren't cleaning up after themselves even in the
+    # successful case. Use the RunTest that loops the reactor a couple of
+    # times after the test is done.
+    return AsynchronousDeferredRunTestForBrokenTwisted.make_factory(
+        timeout=timeout)
+
+
+# By default, asynchronous tests are timed out after 2 minutes.
+DEFAULT_ASYNC_TIMEOUT = 120
```

Comment:
testtools doesn't have such a feature, but that could be added. Not 100% sure if it's a feature for a test framework or a feature for a thing that runs a Twisted reactor until a Deferred fires.

Changed to a timedelta.